### PR TITLE
fix: update getProjectColumns calls (ET-270)

### PR DESCRIPTION
**Pull request by johnkim-det**
## Ticket

ET-270

## Description

Primary task is to add `tableType` param usage to `getProjectColumns` calls (and handle new location types). Also includes other columns-related changes:
- Change initial filter state to empty, instead of relying on a `columnName` value that may not be valid
- Make sure excluding columns from filter or sort actions works as intended
- Prefer column name from `getProjectColumns` API response over name from static front-end definition

## Test Plan

On all DataGrid components (`F_ExperimentList`, `FlatRuns`, `Searches`):
- Regression test for filtering, sorting, and column selection 
- Should consistently show same column names across column headers, filter menu, sort menu, and column selection menu.
- When filters are cleared and the Filters menu is opened, it should initially display a condition with an empty dropdown (not populated with "name")
- Banned filter columns ("searcherMetricsVal") should not have filter options in column header menus or show up in the Filter menu dropdown.
- Banned sort columns ("tags", "searcherMetricsVal") should not have sort options in column header menus or show up in the Sort menu dropdown.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by johnkim-det** [comment]
realize that FE test coverage number is below 80%, but it does look like that's mostly due to changes to page components where unit tests may not make as much sense, and also includes non-Flat Runs-specific changes. but definitely willing to add more tests if anyone has thoughts on where to add coverage.

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Mostly looks good except for terminology.

File: webui/react/src/pages/FlatRuns/columns.ts [hunk 1]
```diff
@@ -172,7 +160,7 @@ export const getColumnDefs = ({
       data: { kind: TEXT_CELL },
       kind: GridCellKind.Custom,
     }),
-    title: 'Searcher Description',
+    title: 'Experiment Description',
```
[the remark this answers is not in the document]
Comment (johnkim-det) [reply webui/react/src/pages/FlatRuns/columns.ts]: updating and confirming with piyush about what these should actually be called

**Review by keita-determined — COMMENTED** [review]
lgtm. please wait for Emily's additional review

**Review by AmanuelAaron — APPROVED** [review]
BE changes LGTM

**Pull request merged**